tests/test_utils.py

```diff
@@ -124,6 +137,40 @@ def test_ingest_injection_catalog(self):
         injected_catalog = self.writeable_butler.get(input_dataset_refs[0])
         self.assertTrue(all(self.injection_catalog == injected_catalog))
 
+    def test_consolidate_injected_catalogs(self):
+        catalog_dict = {"g": self.injected_catalog, "r": self.injected_catalog}
+        output_catalog = consolidate_injected_deepCoadd_catalogs(
+            catalog_dict=catalog_dict,
+            skymap=self.skyMap,
+            tract=9,
+            pixel_match_radius=0.1,
+            get_catalogs_from_butler=False,
+            col_ra="ra",
+            col_dec="dec",
+            col_mag="mag",
+            isPatchInnerKey="injected_isPatchInner",
+            isTractInnerKey="injected_isTractInner",
+            isPrimaryKey="injected_isPrimary",
+            injectionKey="injection_flag",
+        )
+        self.assertEqual(len(output_catalog), 30)
+        expected_columns = {
+            "injection_id",
+            "injected_id",
+            "ra",
+            "dec",
+            "source_type",
+            "g_mag",
+            "r_mag",
+            "injection_draw_size",
+            "injection_flag",
+            "injected_isPatchInner",
+            "injected_isTractInner",
+            "injected_isPrimary",
+        }
+        self.assertEqual(set(output_catalog.columns), expected_columns)
+        self.assertEqual(sum(output_catalog["injected_isPrimary"]), 30)
```

Comment:
I still think you need some kind of test of the injection flags here. I.e., in the `setUp`, you set some of the `injected_isTractInner` and/or `injected_isPatchInner` rows to `False`, and modify a few of the `injection_flag` rows to >0, then make sure that the `injected_isPrimary` column sums to the correct value.